core-web/libs/portlets/edit-ema/portlet/src/lib/store/dot-uve.store.ts

```diff
@@ -0,0 +1,110 @@
+import { patchState, signalStore, withMethods, withState } from '@ngrx/signals';
+import { rxMethod } from '@ngrx/signals/rxjs-interop';
+import { pipe, forkJoin, of } from 'rxjs';
+
+import { HttpErrorResponse } from '@angular/common/http';
+import { inject } from '@angular/core';
+import { Router } from '@angular/router';
+
+import { switchMap, shareReplay, tap, catchError, take } from 'rxjs/operators';
+
+import { DotExperimentsService, DotLanguagesService, DotLicenseService } from '@dotcms/data-access';
+import { LoginService } from '@dotcms/dotcms-js';
+
+import { UVEState } from './models';
+
+import { DotPageApiParams, DotPageApiService } from '../services/dot-page-api.service';
+
+export const UVEStore = signalStore(
+    withState<UVEState>({
+        isEnterprise: false,
+        languages: [],
+        page: undefined,
+        currentUser: undefined,
+        experiment: undefined
+    }),
+    withMethods((store) => {
+        const dotPageApiService = inject(DotPageApiService);
+        const dotLanguagesService = inject(DotLanguagesService);
+        const dotLicenseService = inject(DotLicenseService);
+        const loginService = inject(LoginService);
+        const dotExperimentsService = inject(DotExperimentsService);
+        const router = inject(Router);
+
+        return {
```

Comment:
take a look to this
https://ngrx.io/guide/signals/signal-store/custom-store-features

This store I am pritty sure will be big, can cut by feature the parts
```
export const UVEStore = signalStore(
  withFeatureA(),
  withFeatureB(),
  withFeatureC(),
);
```

Reply:
I will cut with features in the way of refactoring the whole state, right now Im focusing on making it work